app/models/api/sample_io.rb

```diff
@@ -69,6 +80,11 @@ def json_root
     if json_attributes['reference_genome'].blank?
       json_attributes['reference_genome'] = nil
     end
+    if CONTROL_DATA_MAPPING.has_key?(json_attributes['control'])
+      json_attributes['control'] = CONTROL_DATA_MAPPING[json_attributes['control']]
```

Comment:
I don't think you actually need to check the key here as missing keys return nil by default:
```
2.5.3 :005 >  CONTROL_DATA_MAPPING = {
2.5.3 :006 >         'not_control' => '0',
2.5.3 :007 >         'control' => '1',
2.5.3 :008 >         'positive_control' => '1',
2.5.3 :009 >         'negative_control' => '1'
2.5.3 :010?>     }
 => {"not_control"=>"0", "control"=>"1", "positive_control"=>"1", "negative_control"=>"1"}
2.5.3 :011 > CONTROL_DATA_MAPPING['control']
 => "1"
2.5.3 :012 > CONTROL_DATA_MAPPING['UNRECOGNIZED']
 => nil
2.5.3 :013 > CONTROL_DATA_MAPPING[nil]
 => nil
```

Reply:
Also tempted to suggest pulling the control value from the sample directly here, and removing the 'control' section from the standard attributes list above. (I think the sample is accessible via _object)

Reply:
OK, wasn't sure if `map_attribute_to_json_attribute` was doing anything special or not

Reply:
another suggestion, to keep all enum keys in same file, if you get the value from sample directly, then maybe this conversion to boolean can happen in the sample class as well, so here could be like json_attributes['control'] = sample.is_boolean_control? and then in sample class make the conversion?